File: core-mvp/src/main/java/ru/surfstudio/android/core/mvp/model/state/LoadState.java
package ru.surfstudio.android.core.mvp.model.state;

public enum LoadState {
    NONE,                   //контент загружен
    MAIN_LOADING,           //прогресс, закрывающий весь контент
    TRANSPARENT_LOADING,    //полупрозрачный прогресс, блокирует весь интерфейс
    ERROR,                  //ошибка загрузки данных
    EMPTY,                  //данных нет
    NOT_FOUND,              //нет данных по заданному фильтру
}

File: template/base-ui/src/main/java/ru/surfstudio/standard/base_ui/placeholder/LoadState.kt
package ru.surfstudio.android.sample.common.ui.base.loadstate

import ru.surfstudio.android.core.mvp.model.state.LoadStateInterface

class NoneLoadState : LoadStateInterface
class ErrorLoadState : LoadStateInterface
class EmptyLoadState : LoadStateInterface
class NotFoundLoadState : LoadStateInterface
class MainLoadingState : LoadStateInterface
class TransparentLoadingState : LoadStateInterface
class NoInternetLoadState : LoadStateInterface

object LoadState {
    val NONE = NoneLoadState()
    val MAIN_LOADING = MainLoadingState()
    val TRANSPARENT_LOADING = TransparentLoadingState()
    val ERROR = ErrorLoadState()
    val EMPTY = EmptyLoadState()
    val NOT_FOUND = NotFoundLoadState()
    val NO_INTERNET = NoInternetLoadState()
}